### ISSUE TITLE
varLib.iup_delta_optimize can produces different outputs when using `cython`

### ISSUE THREAD

**Issue by cmyr** [fonttools/fonttools#3703]
Still digging into this, but feels worth opening an issue at this point:

The following python program produces different output depending on whether or not varLib is using `cython`:

```python
from fontTools.varLib import iup_delta_optimize


def main():
    deltas = [
        (40, 0),
        (24, -104),
        (15, -104),
        (3, -104),
        (-19, -92),
        (-32, -68),
        (0, 0),
        (0, 0),
        (0, 0),
        (0, 0),
    ]
    orig = [
        (206, 0),
        (174, 458),
        (152, 458),
        (124, 458),
        (74, 486),
        (42, 541),
        (0, 0),
        (0, 0),
        (0, 0),
        (0, 0),
    ]

    result = iup_delta_optimize(deltas, orig, [5], 0.5)
    print(result)

if __name__ == "__main__":
    main()
```

with cython:
```
[(40, 0), (24, -104), None, (3, -104), (-19, -92), (-32, -68), None, None, None, None]
```

without cython:
```
[(40, 0), (24, -104), None, None, (-19, -92), (-32, -68), None, None, None, None]
```

I suspect this has something to do with how the cython code is handling the `tolerance` param; a very good clue is that the output is identical if I change the tolerance from `0.5` to `0.50001`.

edit: right, this strongly echoes #3282

**Comment by behdad** [comment]
Ugh. Not again!

**Comment by behdad** [comment]
It sounds like eg. a fused mul/sum or some other result-changing optimization in the Cython compiled code...

**Comment by behdad** [comment]
> edit: right, this strongly echoes #3282

Yeah I was just looking for that...

**Comment by behdad** [comment]
> > edit: right, this strongly echoes #3282
> 
> Yeah I was just looking for that...

Despite the similarity, I think this is just a float-precision issue.

**Comment by cmyr** [comment]
"fixed" in #3704
